### PR TITLE
Test delete entry in remote Infinispan and check removals are replica…

### PR DESCRIPTION
**Pull request by mhajas**
…ted to the second site anyway

- [x] Test logouts
- [ ] Test refreshes

### CODE REVIEW

**Comment by ahus1** [comment]
@mhajas - I tested this a bit today, and it works nicely for the logouts. 

I verified this also with my local setup where I tested last week: The remove event is propagated to the client for a remote Infinispan 15.0.3, even if the key doesn't exist any more. 

When using 14.0.27 which I did locally, I found that the remove events was propagated to the caller only when the entries existed. So it seems that there is a change in behavior between the major versions. Good that we now have a test for this. 

So I assume we want people to use ISPN 15.x when running with KC 25? 

cc: @ryanemerson @pruivo

**Comment by ryanemerson** [comment]
> So I assume we want people to use ISPN 15.x when running with KC 25?

100%

Let's keep the support matrix as small as possible. This will allow us to spend less time on verifying different combinations and more time enhancing the Keycloak experience for users with new improvements/features.

**Comment by ahus1** [comment]
@ryanemerson / @pruivo - I agree with your comments. Could please also think in the direction if the change in behavior could be seen as a regression by others, and that it would be fixed (to the disadvantage of Keycloak) in an upcoming ISPN release?

**Comment by ryanemerson** [comment]
@ahus1 I believe the change in behaviour was introduced by [ISPN-15211](https://issues.redhat.com/browse/ISPN-15211) and is documented in our upgrading guide.

File: provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/LoginLogoutTest.java [hunk 1]
```diff
@@ -80,4 +86,113 @@ public void testRefreshTokenRevocation() throws Exception {
...
+        Map<String, Object> tokensMap = LOAD_BALANCER_KEYCLOAK.passwordGrant(REALM_NAME, CLIENTID, USERNAME, MAIN_PASSWORD);
+
+        // Make sure all ISPNs can see the entry in the cache
+        assertTrue(DC_1.kc().embeddedIspn().cache(SESSIONS).contains((String) tokensMap.get("session_state")));
+        assertTrue(DC_1.ispn().cache(SESSIONS).contains((String) tokensMap.get("session_state")));
+        assertTrue(DC_2.kc().embeddedIspn().cache(SESSIONS).contains((String) tokensMap.get("session_state")));
+        assertTrue(DC_2.ispn().cache(SESSIONS).contains((String) tokensMap.get("session_state")));
+
+        // Remove the session from the remote store in DC1 only
+        try (var close = InfinispanUtils.withBackupDisabled(DC_1.ispn().cache(SESSIONS), DC_2.ispn().siteName())) {
```
Comment (ryanemerson) [inline provision/rosa-cross-dc/keycloak-benchmark-crossdc-tests/src/test/java/org/keycloak/benchmark/crossdc/LoginLogoutTest.java:102]: `var ignored = ` will stop Intellij complaining about the var being unused.

**Comment by ahus1** [comment]
@ryanemerson - thank you for the note that this is an expected behavior. 

Given that we now depend on ISPN 15 specific behavior, this should be reflected in the docs and the migration guide. I created an issue for that: https://github.com/keycloak/keycloak/issues/29750

Could you please pick that one up and create a [minimal] docs PR for this? Thanks!

**Comment by mhajas** [comment]
As agreed we will not provide a test for updates currently so this PR should be ready for reviewing and merging. @ryanemerson @ahus1 Could you please re-review?

**Review by ryanemerson — APPROVED** [review]

**Comment by ahus1** [comment]
Thank you for these tests, they really helped us to understand Infinispan's behavior around this!

**Pull request merged**